lms/services/assignment.py

```diff
@@ -32,6 +34,16 @@ def get_assignment(self, tool_consumer_instance_guid, resource_link_id):
             .one_or_none()
         )
 
+    def get_assignments_for_grouping(self, grouping_id) -> List[Assignment]:
+        """Get all assignments within a specified grouping."""
+
+        return (
+            self._db.query(Assignment)
+            .join(AssignmentGrouping)
+            .join(Grouping, Grouping.id == grouping_id)
+            .all()
+        )
```

Comment:
As we aren't iterating over groups yet we don't need the join and eager loading stuff. As it turns out this was kind of a bless in disguise as how I'd written it didn't work.

The "noise" fixtures in this test really saved my bacon here by showing me that the eager join condition _does not_ apply to the query, so you end-up getting all the the assignments without this explicit join.

I'm not sure now how the two will play together.